src/ui_dialogs.ts

```diff
@@ -119,6 +165,79 @@ export function openEditLinkDialog(column: LinkColumn, $header: d3.Selection<Col
   });
 }
 
+
+// Renderer type change
+export function rendererTypeDialog(column: Column, $header: d3.Selection<Column>) {
+  var rendererType = column.getRendererType();
+  const rendererTypeList = column.getRendererList();
+
+  var popup = makeSortPopup($header, 'Change Visualization </br>', rendererTypeList.map(function (d, i) {
+    return `<input type="radio" name="renderertype" value=${d.type}  ${(rendererType === d.type) ? 'checked' : ''}> ${d.label}<br>`;
+
+  }).join('\n'));
+
+
+  function thiselement() {
+    return this === (<any>d3.event).target;
+
+  }
+
+  var rendererContent = d3.selectAll('input[name="renderertype"]');
+  rendererContent.on('change', function () {
+    var that = this;
+    rendererType = that.value;
+    column.setRendererType(that.value);
+
+  });
+
+  // To detect if the mouse click event is triggered outside the sort dialog
+  d3.select('body').on('click', function () {
+    var outside = rendererContent.filter(thiselement).empty();
+    if (outside) {
+      popup.remove();
+      d3.select(this).on('click', null);
+    }
+  });
+
+}
+
+// Sort  Dialog.
+export function sortDialog(column: MultiValueColumn, $header: d3.Selection<MultiValueColumn>) {
+
+  var rank = column.getUserSortBy();
+  const valueString: any = [Sort[Sort.min], Sort[Sort.max],Sort[Sort.median], Sort[Sort.q1], Sort[Sort.q3]];
+  const sortLabel: any = ['Min', 'Max', 'Median', 'Q1', 'Q3'];
+
+  var popup = makeSortPopup($header, 'Sort By <br>', valueString.map(function (d, i) {
+    return `<input type="radio" name="multivaluesort" value=${d}  ${(rank === d) ? 'checked' : ''} > ${sortLabel[i]} <br>`;
+
+  }).join('\n'));
+
+  function thiselement() {
+    return this === (<any>d3.event).target;
+  }
+
+  // To detect if the mouse click event is triggered outside the sort dialog
+  var sortContent = d3.selectAll('input[name=multivaluesort]');
+  sortContent.on('change', function () {
+    var that = this;
+    rank = that.value;
+    column.setUserSortBy(rank);
+
+  });
+
+  // To detect if the mouse click event is triggered outside the sort dialog
+  d3.select('body').on('click', function () {
```

Comment:
code duplication to the previous method that should be avoided, especially since it is complex code. Try to refactor it, e.g. extracting a helper method